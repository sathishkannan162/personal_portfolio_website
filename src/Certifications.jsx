import { Heading, SimpleGrid } from '@chakra-ui/react';
import CertificateCard from './CertificateCard';
import PageContainer from './PageContainer';

export default function Certifications() {
  return (
    <PageContainer>
      <Heading>Certifications</Heading>
      <SimpleGrid py={4} columns={[1, 2, 3]} spacing="15px">
        <CertificateCard />
      </SimpleGrid>
    </PageContainer>
  );
}
