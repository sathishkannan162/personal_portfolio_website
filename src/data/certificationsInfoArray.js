const certificationsInfoArray = [
  {
    certificate_ID: 1,
    name: 'Responsive Web Design',
    provider: 'FreeCodeCamp',
    imageUrl: 'https://projectimages.netlify.app/certificates/certificate_1.jpeg',
    certificateUrl:
      'https://www.freecodecamp.org/certification/sathish_kannan/responsive-web-design',
  },
  {
    certificate_ID: 2,
    name: 'JavaScript Algorithms and Data Structures',
    provider: 'FreeCodeCamp',
    imageUrl: 'https://projectimages.netlify.app/certificates/certificate_2.jpeg',
    certificateUrl:
      'https://www.freecodecamp.org/certification/sathish_kannan/javascript-algorithms-and-data-structures',
  },
  {
    certificate_ID: 3,
    name: 'Front End Development Libraries',
    provider: 'FreeCodeCamp',
    imageUrl: 'https://projectimages.netlify.app/certificates/certificate_3.jpeg',
    certificateUrl:
      'https://www.freecodecamp.org/certification/sathish_kannan/front-end-development-libraries',
  },
  {
    certificate_ID: 4,
    name: 'Data Visualization',
    provider: 'FreeCodeCamp',
    imageUrl: 'https://projectimages.netlify.app/certificates/certificate_4.jpeg',
    certificateUrl:
      'https://www.freecodecamp.org/certification/sathish_kannan/data-visualization',
  },
  {
    certificate_ID: 5,
    name: 'Relational Database',
    provider: 'FreeCodeCamp',
    imageUrl: 'https://projectimages.netlify.app/certificates/certificate_7.jpeg',
    certificateUrl:
      'https://www.freecodecamp.org/certification/sathish_kannan/relational-database-v8',
  },
  {
    certificate_ID: 6,
    name: 'Back End Development and APIs',
    provider: 'FreeCodeCamp',
    imageUrl: 'https://projectimages.netlify.app/certificates/certificate_5.jpeg',
    certificateUrl:
      'https://www.freecodecamp.org/certification/sathish_kannan/back-end-development-and-apis',
  },
  {
    certificate_ID: 7,
    name: 'Quality Assurance',
    provider: 'FreeCodeCamp',
    imageUrl: 'https://projectimages.netlify.app/certificates/certificate_6.jpeg',
    certificateUrl:
      'https://www.freecodecamp.org/certification/sathish_kannan/quality-assurance-v7',
  },
];

export default certificationsInfoArray.reverse();
