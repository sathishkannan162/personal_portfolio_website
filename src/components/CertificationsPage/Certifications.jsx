import { Heading, SimpleGrid, Box } from '@chakra-ui/react';
import CertificateCard from './CertificateCard';
import PageContainer from '../CommonComponents/PageContainer';
import certificationsInfoArray from '../../data/certificationsInfoArray.js';

export default function Certifications() {
  return (
    <PageContainer>
      <Heading>Certifications</Heading>
      <SimpleGrid py={4} columns={[1, 1, 2]} spacing="35px">
        {certificationsInfoArray.map((certificate) => {
          return (
            <Box key={certificate.certificate_ID}>
              <CertificateCard certificate={certificate} />
            </Box>
          );
        })}
      </SimpleGrid>
    </PageContainer>
  );
}
