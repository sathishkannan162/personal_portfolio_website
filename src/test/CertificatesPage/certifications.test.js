import React from 'react';
import { render, screen } from '../test-utils';
import Certifications from '../../components/CertificationsPage/Certifications';
import certificationsInfoArray from '../../data/certificationsInfoArray';

class IntersectionObserver {
  // constructor(callback, options) {}
  constructor(callback, options) {
    this.callback = callback;
    this.options = options;
  }
  observe(target) {
    this.target = target;
    this.callback([
      {
        isIntersecting: true,
        target: this.target,
      },
    ]);
  }

  unobserve() { }
}

window.IntersectionObserver = IntersectionObserver;
describe('certifications page', () => {
test('renders Certifications component correctly', () => {
  render(
    <Certifications />
  );

  // Verify that the heading is rendered
  const headingElement = screen.getByRole('heading', { name: /certifications/i });
  expect(headingElement).toBeInTheDocument();

  // Verify that the CertificateCard components are rendered
  const certificateCardElements = screen.getAllByTestId('certificate-card');
  expect(certificateCardElements).toHaveLength(certificationsInfoArray.length);
});
    test('captures and compares component snapshot', () => {
        const { asFragment } = render(<Certifications />);
        expect(asFragment()).toMatchSnapshot();
    });

})
