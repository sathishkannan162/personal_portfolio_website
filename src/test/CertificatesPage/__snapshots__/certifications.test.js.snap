// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`certifications page captures and compares component snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1qqblot"
  >
    <h2
      class="chakra-heading css-1dklj6k"
    >
      Certifications
    </h2>
    <div
      class="css-qpptq9"
    >
      <div
        class="css-0"
      >
        <div
          class="css-0"
          data-test="certificate-card"
        >
          <h2
            class="chakra-heading css-sj68pu"
          >
            Responsive Web Design
          </h2>
          <div
            data-test="certificate-div"
            style="opacity: 0; transform: translateY(20px) translateZ(0);"
            tabindex="0"
          >
            <a
              class="chakra-link css-spn4bz"
              href="https://www.freecodecamp.org/certification/sathish_kannan/responsive-web-design"
              target="_blank"
            >
              <div
                class="css-101iv6h"
              >
                <div
                  class="css-88irkg"
                >
                  <img
                    alt="placeholder image"
                    class="chakra-image css-q7lffx"
                    src="https://projectimages.netlify.app/certificates/certificate_1.jpeg"
                  />
                </div>
                <div>
                  <div
                    class="css-iw9ipp"
                  >
                    <h2
                      class="chakra-heading css-c7nxzi"
                    >
                      <p
                        class="chakra-text css-1oj1yll"
                      >
                        Responsive Web Design
                      </p>
                      <p
                        class="chakra-text css-1rhvqo5"
                      >
                        FreeCodeCamp
                      </p>
                      <p
                        class="chakra-text css-5t65gu"
                      >
                        Click Card to View
                      </p>
                    </h2>
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        class="css-0"
      >
        <div
          class="css-0"
          data-test="certificate-card"
        >
          <h2
            class="chakra-heading css-sj68pu"
          >
            JavaScript Algorithms and Data Structures
          </h2>
          <div
            data-test="certificate-div"
            style="opacity: 0; transform: translateY(20px) translateZ(0);"
            tabindex="0"
          >
            <a
              class="chakra-link css-spn4bz"
              href="https://www.freecodecamp.org/certification/sathish_kannan/javascript-algorithms-and-data-structures"
              target="_blank"
            >
              <div
                class="css-8v8vvc"
              >
                <div
                  class="css-88irkg"
                >
                  <img
                    alt="placeholder image"
                    class="chakra-image css-q7lffx"
                    src="https://projectimages.netlify.app/certificates/certificate_1.jpeg"
                  />
                </div>
                <div>
                  <div
                    class="css-iw9ipp"
                  >
                    <h2
                      class="chakra-heading css-c7nxzi"
                    >
                      <p
                        class="chakra-text css-1oj1yll"
                      >
                        JavaScript Algorithms and Data Structures
                      </p>
                      <p
                        class="chakra-text css-1rhvqo5"
                      >
                        FreeCodeCamp
                      </p>
                      <p
                        class="chakra-text css-5t65gu"
                      >
                        Click Card to View
                      </p>
                    </h2>
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        class="css-0"
      >
        <div
          class="css-0"
          data-test="certificate-card"
        >
          <h2
            class="chakra-heading css-sj68pu"
          >
            Front End Development Libraries
          </h2>
          <div
            data-test="certificate-div"
            style="opacity: 0; transform: translateY(20px) translateZ(0);"
            tabindex="0"
          >
            <a
              class="chakra-link css-spn4bz"
              href="https://www.freecodecamp.org/certification/sathish_kannan/front-end-development-libraries"
              target="_blank"
            >
              <div
                class="css-uco9q9"
              >
                <div
                  class="css-88irkg"
                >
                  <img
                    alt="placeholder image"
                    class="chakra-image css-q7lffx"
                    src="https://projectimages.netlify.app/certificates/certificate_1.jpeg"
                  />
                </div>
                <div>
                  <div
                    class="css-iw9ipp"
                  >
                    <h2
                      class="chakra-heading css-c7nxzi"
                    >
                      <p
                        class="chakra-text css-1oj1yll"
                      >
                        Front End Development Libraries
                      </p>
                      <p
                        class="chakra-text css-1rhvqo5"
                      >
                        FreeCodeCamp
                      </p>
                      <p
                        class="chakra-text css-5t65gu"
                      >
                        Click Card to View
                      </p>
                    </h2>
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        class="css-0"
      >
        <div
          class="css-0"
          data-test="certificate-card"
        >
          <h2
            class="chakra-heading css-sj68pu"
          >
            Data Visualization
          </h2>
          <div
            data-test="certificate-div"
            style="opacity: 0; transform: translateY(20px) translateZ(0);"
            tabindex="0"
          >
            <a
              class="chakra-link css-spn4bz"
              href="https://www.freecodecamp.org/certification/sathish_kannan/data-visualization"
              target="_blank"
            >
              <div
                class="css-1q3zrop"
              >
                <div
                  class="css-88irkg"
                >
                  <img
                    alt="placeholder image"
                    class="chakra-image css-q7lffx"
                    src="https://projectimages.netlify.app/certificates/certificate_1.jpeg"
                  />
                </div>
                <div>
                  <div
                    class="css-iw9ipp"
                  >
                    <h2
                      class="chakra-heading css-c7nxzi"
                    >
                      <p
                        class="chakra-text css-1oj1yll"
                      >
                        Data Visualization
                      </p>
                      <p
                        class="chakra-text css-1rhvqo5"
                      >
                        FreeCodeCamp
                      </p>
                      <p
                        class="chakra-text css-5t65gu"
                      >
                        Click Card to View
                      </p>
                    </h2>
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        class="css-0"
      >
        <div
          class="css-0"
          data-test="certificate-card"
        >
          <h2
            class="chakra-heading css-sj68pu"
          >
            Back End Development and APIs
          </h2>
          <div
            data-test="certificate-div"
            style="opacity: 0; transform: translateY(20px) translateZ(0);"
            tabindex="0"
          >
            <a
              class="chakra-link css-spn4bz"
              href="https://www.freecodecamp.org/certification/sathish_kannan/back-end-development-and-apis"
              target="_blank"
            >
              <div
                class="css-tx2gkx"
              >
                <div
                  class="css-88irkg"
                >
                  <img
                    alt="placeholder image"
                    class="chakra-image css-q7lffx"
                    src="https://projectimages.netlify.app/certificates/certificate_1.jpeg"
                  />
                </div>
                <div>
                  <div
                    class="css-iw9ipp"
                  >
                    <h2
                      class="chakra-heading css-c7nxzi"
                    >
                      <p
                        class="chakra-text css-1oj1yll"
                      >
                        Back End Development and APIs
                      </p>
                      <p
                        class="chakra-text css-1rhvqo5"
                      >
                        FreeCodeCamp
                      </p>
                      <p
                        class="chakra-text css-5t65gu"
                      >
                        Click Card to View
                      </p>
                    </h2>
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        class="css-0"
      >
        <div
          class="css-0"
          data-test="certificate-card"
        >
          <h2
            class="chakra-heading css-sj68pu"
          >
            Quality Assurance
          </h2>
          <div
            data-test="certificate-div"
            style="opacity: 0; transform: translateY(20px) translateZ(0);"
            tabindex="0"
          >
            <a
              class="chakra-link css-spn4bz"
              href="https://www.freecodecamp.org/certification/sathish_kannan/quality-assurance-v7"
              target="_blank"
            >
              <div
                class="css-8ytn6q"
              >
                <div
                  class="css-88irkg"
                >
                  <img
                    alt="placeholder image"
                    class="chakra-image css-q7lffx"
                    src="https://projectimages.netlify.app/certificates/certificate_1.jpeg"
                  />
                </div>
                <div>
                  <div
                    class="css-iw9ipp"
                  >
                    <h2
                      class="chakra-heading css-c7nxzi"
                    >
                      <p
                        class="chakra-text css-1oj1yll"
                      >
                        Quality Assurance
                      </p>
                      <p
                        class="chakra-text css-1rhvqo5"
                      >
                        FreeCodeCamp
                      </p>
                      <p
                        class="chakra-text css-5t65gu"
                      >
                        Click Card to View
                      </p>
                    </h2>
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
      <div
        class="css-0"
      >
        <div
          class="css-0"
          data-test="certificate-card"
        >
          <h2
            class="chakra-heading css-sj68pu"
          >
            Relational Database
          </h2>
          <div
            data-test="certificate-div"
            style="opacity: 0; transform: translateY(20px) translateZ(0);"
            tabindex="0"
          >
            <a
              class="chakra-link css-spn4bz"
              href="https://www.freecodecamp.org/certification/sathish_kannan/relational-database-v8"
              target="_blank"
            >
              <div
                class="css-1mnit0y"
              >
                <div
                  class="css-88irkg"
                >
                  <img
                    alt="placeholder image"
                    class="chakra-image css-q7lffx"
                    src="https://projectimages.netlify.app/certificates/certificate_1.jpeg"
                  />
                </div>
                <div>
                  <div
                    class="css-iw9ipp"
                  >
                    <h2
                      class="chakra-heading css-c7nxzi"
                    >
                      <p
                        class="chakra-text css-1oj1yll"
                      >
                        Relational Database
                      </p>
                      <p
                        class="chakra-text css-1rhvqo5"
                      >
                        FreeCodeCamp
                      </p>
                      <p
                        class="chakra-text css-5t65gu"
                      >
                        Click Card to View
                      </p>
                    </h2>
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
